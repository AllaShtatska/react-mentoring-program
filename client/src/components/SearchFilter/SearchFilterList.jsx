import React from 'react';
import PropTypes from 'prop-types';
import SEARCH_FILTER_TYPE from './searchFilterType'
import SearchFilter from './SearchFilter'


const SearchFilterList = ({ filter, onSwitchFilter }) => {
    const searchFilters = Object.values(SEARCH_FILTER_TYPE).map(searchFilter => (
      <SearchFilter
        key={searchFilter}
        currentFilter={filter}
        filter={searchFilter}
        onSwitchFilter={onSwitchFilter}
     />
    ));
    
    return(
      <div className="inline">
         Search by
        { searchFilters }
      </div>
    );
  };

  SearchFilterList.propTypes = {
    filter: PropTypes.string.isRequired,
    onSwitchFilter: PropTypes.func.isRequired,
  };

export default SearchFilterList;