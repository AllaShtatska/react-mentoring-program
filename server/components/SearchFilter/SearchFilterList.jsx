import React from "react";
import PropTypes from "prop-types";

import SEARCH_FILTER_TYPE from "./searchFilterType";
import SearchFilter from "./SearchFilter";
import inlineStyles from "../../static/inline.css";

const SearchFilterList = ({ filter, onSwitchFilter }) => {
  const searchFilters = Object.values(SEARCH_FILTER_TYPE).map(searchFilter => (
    <SearchFilter
      key={searchFilter}
      currentFilter={filter}
      filter={searchFilter}
      onSwitchFilter={onSwitchFilter}
    />
  ));

  return (
    <div className={inlineStyles.inline}>
      Search by
      {searchFilters}
    </div>
  );
};

SearchFilterList.propTypes = {
  filter: PropTypes.string.isRequired,
  onSwitchFilter: PropTypes.func.isRequired
};

export default SearchFilterList;
