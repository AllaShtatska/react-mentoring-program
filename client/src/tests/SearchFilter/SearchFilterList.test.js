import React from 'react';
import renderer from 'react-test-renderer';
import SearchFilterList from '../../components/SearchFilter/SearchFilterList';
import SEARCH_FILTER_TYPE from '../../components/SearchFilter/searchFilterType';

it ('renders correctly', ()=>{
    const component = renderer.create(<SearchFilterList filter={SEARCH_FILTER_TYPE.title} onSwitchFilter={function(){}}/>);

    var jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();
});