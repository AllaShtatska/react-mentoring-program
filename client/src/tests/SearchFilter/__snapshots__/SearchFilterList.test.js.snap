// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="inline"
>
  Search by
  <div
    className="inline"
  >
    <button
      className="filter-button-selected"
      onClick={[Function]}
    >
      Title
    </button>
  </div>
  <div
    className="inline"
  >
    <button
      className="filter-button-deselected"
      onClick={[Function]}
    >
      Genre
    </button>
  </div>
</div>
`;
